### PR TITLE
For combo_handler plugin, add an optional whitelist of allowed values for Content-Type.

### PR DESCRIPTION
**Pull request by ywkaras**
The whitelist is in a text file, whose name is given on the plugin line in plugin.config.

### CODE REVIEW

File: plugins/esi/combo_handler.cc [hunk 1]
```diff
@@ -1052,21 +1095,86 @@ getContentType(TSMBuffer bufp, TSMLoc hdr_loc, string &resp_header_fields)
...
+
+  fs.open(file_spec);
+  if (fs.good()) {
+    for (;;) {
+      ++line_num;
+      fs.getline(line_buffer, sizeof(line_buffer));
+      if (!fs.good()) {
+        break;
+      }
+      std::string_view bs{" \t"sv};
```
Comment (SolidWallOfCode) [inline plugins/esi/combo_handler.cc]: You use this several places, you should make it a `constexpr` value to ensure consistency.

File: plugins/esi/combo_handler.cc [hunk 2]
```diff
@@ -1052,21 +1095,86 @@ getContentType(TSMBuffer bufp, TSMLoc hdr_loc, string &resp_header_fields)
...
+      ts::TextView line{line_buffer, std::size_t(fs.gcount() - 1)};
+      line.ltrim(bs);
+      if (line[0] == '#') {
```
Comment (SolidWallOfCode) [inline plugins/esi/combo_handler.cc]: Might as well do `line.empty() || line[0] == '#'`.

File: doc/admin-guide/plugins/combo_handler.en.rst [hunk 3]
```diff
@@ -47,6 +47,15 @@ The arguments in the :file:`plugin.config` line in order represent
 2. The name of the key used for signature verification (disabled by
    default)
 
+3. A colon separated list of headers which, if present on at least one response, will be
+   added to the combo response.
+
+4. The path of a config file with allowed content types of objects to be combined, one per
+   line, witout parameters. Parameters in the Content-Type field value will be ignored when
```
Comment (SolidWallOfCode) [inline doc/admin-guide/plugins/combo_handler.en.rst]: "without"

File: plugins/esi/combo_handler.cc [hunk 4]
```diff
@@ -1052,21 +1095,86 @@ getContentType(TSMBuffer bufp, TSMLoc hdr_loc, string &resp_header_fields)
...
     for (int i = 0; i < n_values; ++i) {
       value = TSMimeHdrFieldValueStringGet(bufp, hdr_loc, field_loc, i, &value_len);
-      if (!values_added) {
-        resp_header_fields.append("Content-Type: ");
-        values_added = true;
-      } else {
-        resp_header_fields.append(", ");
+      ts::TextView tv{value, value_len};
+      tv              = tv.prefix(';').rtrim(std::string_view(" \t"));
+      auto const &ctw = _content_type_whitelist;
```
Comment (SolidWallOfCode) [inline plugins/esi/combo_handler.cc]: What is the point of `ctw`? Why not use `_content_type_whitelist`?

File: plugins/esi/combo_handler.cc [hunk 5]
```diff
@@ -1052,21 +1095,86 @@ getContentType(TSMBuffer bufp, TSMLoc hdr_loc, string &resp_header_fields)
...
+        // Comment line.
+        continue;
+      }
+      ts::TextView content_type{line.take_prefix_at(bs)};
+      line.trim(bs);
+      if (line.size() && (line[0] != '#')) {
+        extra_junk_on_line = true;
+        break;
+      }
+      _content_type_whitelist.emplace_back(std::string_view(content_type));
```
Comment (SolidWallOfCode) [inline plugins/esi/combo_handler.cc]: Do you need the cast to `string_view` here?

File: plugins/esi/combo_handler.cc (hunk 4 above)
Comment (ywkaras) [reply plugins/esi/combo_handler.cc]: You got it SWOV.

**Comment by ywkaras** [comment]
OK updated.

File: plugins/esi/combo_handler.cc (hunk 5 above)
Comment (ywkaras) [reply plugins/esi/combo_handler.cc]: Apparently not.

**Comment by zwoop** [comment]
Cherry-picked to v9.0.x branch.

**Pull request merged**